### PR TITLE
Apply event sourcing to the incident create and created processors

### PR DESCRIPTION
**Pull request by korthout**
## Description

- introduces `IncidentCreatedEventApplier`
- removes state changes from `CreateIncidentProcesssor`
- migrates `Create` and `Created` Incident records

For the reviewer: 
- I've split up the commits into 2 commits: 1 for refactoring and 1 for the logic change
- this does not yet close the issue, incident:resolve and resolved also need to be migrated still

## Related issues

relates #6174 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by korthout** [comment]
bors try

File: engine/src/main/java/io/zeebe/engine/processing/incident/CreateIncidentProcessor.java [hunk 1]
```diff
@@ -41,51 +45,52 @@ public boolean onCommand(
       final CommandControl<IncidentRecord> commandControl) {
     final IncidentRecord incidentEvent = command.getValue();
 
-    final boolean incidentIsNotRejected = !rejectIncidentCreation(incidentEvent, commandControl);
+    final boolean incidentIsNotRejected = !tryRejectIncidentCreation(incidentEvent, commandControl);
 
     if (incidentIsNotRejected) {
-      final long incidentKey = commandControl.accept(IncidentIntent.CREATED, incidentEvent);
-      zeebeState.getIncidentState().createIncident(incidentKey, incidentEvent);
+      commandControl.accept(IncidentIntent.CREATED, incidentEvent);
```
Comment (Zelldon) [inline engine/src/main/java/io/zeebe/engine/processing/incident/CreateIncidentProcessor.java:51]: Nothing we need to do now but just thought we could also invert the logic, like rename method to createIncident, which can return true or false and on false it means we are not able to create an incident and reject it.

**Review by ChrisKujawa — APPROVED** [review]
thanks good work :medal_military:

File: engine/src/main/java/io/zeebe/engine/processing/incident/CreateIncidentProcessor.java (hunk 1 above)
Comment (korthout) [reply engine/src/main/java/io/zeebe/engine/processing/incident/CreateIncidentProcessor.java:51]: Agreed, but we should probably also unimplement this as a CommandProcessor, might be nice cleanup work :)

**Comment by korthout** [comment]
bors merge

**Pull request merged**